### PR TITLE
enh(api): Add API endpoint to list host severities

### PR DESCRIPTION
**Pull request by loiclau**
## Description

Add API endpoint to list host severities

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

GET on /centreon/api/v2.1/configuration/hosts/severities

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            LEFT JOIN `centreon`.view_img_dir_relation iconR
+                ON iconR.img_img_id = icon.img_id
+            LEFT JOIN `centreon`.view_img_dir iconD
+                ON iconD.dir_id = iconR.dir_dir_parent_id'
+        );
+
+        // Search
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+        $request .= !is_null($searchRequest)
+            ? $searchRequest . ' AND level IS NOT NULL'
```
Comment (callapa) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php:97]: You cannot allow the filtering of the 'level' property and then force its value in the query.
Please test all search parameters.

**Review by callapa — CHANGES_REQUESTED** [review]
Don't forget to change the date of the copyright in all PHP files.

File: src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    /**
+     * @param int $level
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setLevel(int $level): HostSeverity
+    {
+        if ($level !== null) {
+            Assertion::maxLength((string) $level, self::MAX_LEVEL_LENGTH, 'HostSeverity::level');
```
Comment (callapa) [inline src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php:186]: Don't use `maxLength` assert, use `max` instead.
Don't forget to add the `min` assert also

File: src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    public function isActivated(): bool
+    {
+        return $this->isActivated;
+    }
+
+    /**
+     * @param bool $isActivated
+     * @return HostSeverity
+     */
+    public function setIsActivated(bool $isActivated): HostSeverity
```
Comment (callapa) [inline src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php:164]:
```suggestion
    public function setActivated(bool $isActivated): HostSeverity
```

File: src/Centreon/Infrastructure/HostConfiguration/API/Model/HostSeverityV21.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public $level;
+
+    /**
+     * @var string
+     */
+    public $icon;
+
+    /**
+     * @var string
```
Comment (callapa) [inline src/Centreon/Infrastructure/HostConfiguration/API/Model/HostSeverityV21.php:58]:
```suggestion
     * @var string|null
```

**Review by callapa — CHANGES_REQUESTED** [review]
Please correct the date in all copyrights.

**Review by callapa — DISMISSED** [review]

File: tests/api/behat.yml [hunk 5]
```diff
@@ -52,6 +56,7 @@ default:
       paths: [ "%paths.base%/features/MonitoringTimeline.feature" ]
       contexts:
         - Centreon\Test\Api\Context\MonitoringTimelineContext
+
```
Comment (loiclau) [inline tests/api/behat.yml:59]:
```suggestion
```

**Review by callapa — DISMISSED** [review]

**Review by kduret — COMMENTED** [review]
You need to use iconUrlNormalizer in controller : https://github.com/centreon/centreon/blob/master/src/Centreon/Application/Normalizer/IconUrlNormalizer.php
If not, icon path is not readable by the frontend

File: src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+    /**
+     * @param string $name
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public function setName(string $name): HostSeverity
+    {
+        Assertion::maxLength($name, self::MAX_NAME_LENGTH, 'HostSeverity::name');
+        $this->name = $name;
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/Model/HostSeverity.php:125]: shouldn't the name be not empty too ?

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                    if (is_bool($valueToNormalize)) {
+                        return ($valueToNormalize === true) ? '1' : '0';
+                    }
+                    return $valueToNormalize;
+                }
+            }
+        );
+        $request = $this->translateDbName(
+            'SELECT SQL_CALC_FOUND_ROWS hc.*, icon.img_id AS img_id, icon.img_name AS img_name,
+                CONCAT(iconD.dir_name,\'/\',icon.img_path) AS img_path, icon.img_comment AS img_comment
```
Comment (sc979) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php:83]: You used the DIRECTORY_SEPARATOR here : 
https://github.com/centreon/centreon/pull/9371/files#diff-0b917d5735d4715bfbe78685b4a9d705f9aefbe58c3b7e4d7e61d86e0b3442d4R101
Meaning that / or \ can be used
```suggestion
                CONCAT(iconD.dir_name,\'' . DIRECTORY_SEPARATOR . '\',icon.img_path) AS img_path, icon.img_comment AS img_comment
```

File: src/Centreon/Infrastructure/HostConfiguration/Repository/Model/HostSeverityFactoryRdb.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+     * @return HostSeverity
+     * @throws \Assert\AssertionFailedException
+     */
+    public static function create(array $data): HostSeverity
+    {
+        $icon = (new Image())
+            ->setId((int)$data['img_id'])
+            ->setName($data['img_name'])
+            ->setComment($data['img_comment'])
+            ->setPath(str_replace('//', '/', ($data['img_path'])));
```
Comment (sc979) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/Model/HostSeverityFactoryRdb.php:48]: Souldn't you use DIRECTORY_SEPARATOR instead ?

File: tests/api/features/HostSeverity.feature [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        ],
+        "meta": {
+            "page": 1,
+            "limit": 10,
+            "search": {},
+            "sort_by": {},
+            "total": 1
+        }
+    }
+    """
```
Comment (sc979) [inline tests/api/features/HostSeverity.feature:49]:
```suggestion
    """

```

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php (hunk 7 above)
Comment (loiclau) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/HostSeverityRepositoryRDB.php:83]: path is for web, not for system

File: src/Centreon/Infrastructure/HostConfiguration/Repository/Model/HostSeverityFactoryRdb.php (hunk 8 above)
Comment (loiclau) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/Model/HostSeverityFactoryRdb.php:48]: path is for web, not for system

**Review by callapa — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**